src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,140 +1,381 @@
-import java.util.Iterator;
-import java.util.List;
-import java.util.ListIterator;
+import java.util.*;
 
 public class StudentList implements List<Student> {
+    private int capacity;
+    private int size = 0;
+    private Student[] array;
 
+    public StudentList() {
+        this.capacity = 10;
+        this.array = new Student[this.capacity];
+    }
+
+    public StudentList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.capacity = initialCapacity;
+            this.array = new Student[this.capacity];
+        } else {
+            if (initialCapacity != 0) {
+                throw new IllegalArgumentException();
+            }
+
+            this.capacity = 10;
+            this.array = new Student[this.capacity];
+        }
+
+    }
+
+    public StudentList(Collection<? extends Student> collection) {
+        if (collection == null) {
+            throw new NullPointerException();
+        } else {
+            this.size = collection.size();
+            this.capacity = this.size * 2;
+            this.array = new Student[this.capacity];
+            Iterator var2 = collection.iterator();
+
+            while(var2.hasNext()) {
+                Student t = (Student)var2.next();
+                this.add(t);
+            }
+
+        }
+    }
     @Override
     public int size() {
-        //TODO
-        return 0;
+        return this.size;
     }
 
     @Override
     public boolean isEmpty() {
-        //TODO
-        return false;
+        return this.size == 0;
     }
 
     @Override
     public boolean contains(Object o) {
-        //TODO
-        return false;
+        boolean flag = false;
+
+        for(int i = 0; i < this.size; i++) {
+            if (o.equals(this.array[i])) {
+                flag = true;
+                break;
+            }
+        }
+
+        return flag;
     }
 
     @Override
     public Iterator<Student> iterator() {
-        //TODO
-        return null;
+        return listIterator(0);
     }
 
     @Override
     public Object[] toArray() {
-        //TODO
-        return null;
+        return Arrays.copyOf(this.array, this.size);
     }
 
     @Override
     public <T> T[] toArray(T[] ts) {
-        //TODO
-        return null;
+        return (T[])Arrays.copyOf(this.array,this.size);
     }
 
     @Override
     public boolean add(Student student) {
-        //TODO
+        if ((size + size/2)>=capacity) {
+            this.capacity *= 2;
+        }
+
+        ++this.size;
+        Student[] arr = new Student[this.capacity];
+
+        for(int i = 0; i < this.size - 1; ++i) {
+            arr[i] = this.array[i];
+        }
+
+        arr[this.size - 1] = student;
+        this.array = arr;
         return false;
     }
 
     @Override
     public boolean remove(Object o) {
-        //TODO
-        return false;
+        boolean flag = false;
+        if (o != null) {
+            for(int i = 0; i < this.size; ++i) {
+                if (o.equals(this.array[i])) {
```

Comment:
Object can be null (`NullPointerException` vulnerability)

Reply:
Fixed!